boards/arm/efm32hg_slstk3400a/efm32hg_slstk3400a.dts

```diff
@@ -101,3 +112,12 @@
 
 	};
 };
+
+&timer0 {
+	status = "okay";
+
+	pwm0: pwm {
+		status = "okay";
+		pin-location = <GECKO_LOCATION(0) GECKO_PORT_A GECKO_PIN(0)>;
```

Comment:
Hi! Thanks for your contribution. Unfortunately, this way of setting up pinmuxing is considered deprecated in Zephyr. Please consult with "this page":https://docs.zephyrproject.org/latest/hardware/pinctrl/index.html to see how to use the Pinctrl API that is used in Zephyr for pinmuxing.

Reply:
Hi! I started working on the requested change. I pushed the progress I have so far, but the build is unstable for the time being. It fails while generating the updated configuration for the `pwm_gecko` driver, that should contain the `pinctrl_dev_config`.  I will get back to it when I can spare some time. I would appreciate if you could take a look at it in the meantime.

PS: Probably a ton of things will go wrong with the CI/CD checks, sorry for that in advance, I only published the changes so you can take a look at it.